test/any/URLExtractor-test.js

```diff
@@ -0,0 +1,130 @@
+// Copyright 2017 The Chromium Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+'use strict';
+
+describe('URL Extractor', function() {
+  var URLExtractor;
+  var IDLFileContents;
+
+  beforeEach(function() {
+    URLExtractor = foam.lookup('org.chromium.webidl.URLExtractor');
+    IDLFileContents = foam.lookup('org.chromium.webidl.IDLFileContents');
+  });
+
+  it('should return undefined for passing incorrect parameter type', function() {
+    var extractor = URLExtractor.create();
+    var results = extractor.extract('Test content');
+    expect(results).toBeUndefined();
+  });
+
+  it('should return no URLs', function() {
+    var extractor = URLExtractor.create();
+    var file = IDLFileContents.create({
+      contents: 'Test content',
+    });
+
+    var results = extractor.extract(file);
+    expect(results).toBeDefined();
+    expect(results.length).toBe(0);
+  });
+
+  it('should return all HTTP(s) URLs within the text fragment', function() {
+    var extractor = URLExtractor.create();
+    var urls = [ 'http://www.google.com', 'https://www.google.com/',
+        'ftp://www.google.com', 'http://google.com', 'https://google.com',
+        'https://google.com/potato?hello=true', 'www.google.com' ];
```

Comment:
The pattern should only match up to the anchor tag. Everything after the anchor tag should be dropped (the anchor doesn't matter since the entire page is fetched anyways). I will add a test in the next commit to verify this.